src/Microsoft.Extensions.Hosting/Internal/ConsoleLifetime.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Hosting.Internal
+{
+    /// <summary>
+    /// Listens for Ctrl+C or SIGTERM and initiates shutdown.
+    /// </summary>
+    public class ConsoleLifetime : IHostLifetime
+    {
+        public ConsoleLifetime(IHostingEnvironment environment, IApplicationLifetime applicationLifetime)
+        {
+            Environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            ApplicationLifetime = applicationLifetime ?? throw new ArgumentNullException(nameof(applicationLifetime));
+            ApplicationLifetime.ApplicationStarted.Register(() =>
+            {
+                Console.WriteLine("Application starting. Press Ctrl+C to shut down.");
```

Comment:
I dunno if we should repeat the same mistake here... Is this the only reason we wire up the event 😄

Reply:
At least now it's contained in the ConsoleLifetime instance. It can now have its own options, or just be replaced.

Reply:
We need to add this back as the default lifetime implementation:

https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs#L137

Process exit.